Mathlib/CategoryTheory/Widesubcategory.lean

```diff
@@ -0,0 +1,125 @@
+/-
+Copyright (c) 2024 Sina Hazratpour. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sina Hazratpour
+-/
+import Mathlib.CategoryTheory.Functor.FullyFaithful
+import Mathlib.CategoryTheory.MorphismProperty.Composition
+
+/-!
+# Wide subcategories
+
+A wide subcategory of a category `C` is a subcategory containing all the objects of `C`.
+
+## Main declarations
+
+Given a category `D`, a function `F : C → D `from a type `C` to the
+objects of `D`, a morphism property `P` on `D` which contains identities and is stable under
+composition, witnessed by `hP`, the type class `InducedWideCategory D F P hP` is a typeclass
+synonym for `C` which comes equipped with a category structure with morphisms `X ⟶ Y` being the
+morphisms in `D` which have the property `P`.
+
+The instance `WideSubcategory.category` provides a category structure on `WideSubcategory P hP`
+whose objects are the objects of `C` and morphisms are the morphisms in `C` which have the
+property `P`.
+-/
+
+namespace CategoryTheory
+
+universe v₁ v₂ u₁ u₂
+
+open MorphismProperty
+
+section Induced
+
+variable {C : Type u₁} (D : Type u₂) [Category.{v₁} D]
+variable (F : C → D) (P : MorphismProperty D)
+    (hP : IsMultiplicative P)
```

Comment:
`IsMultiplicative` is a type-class:
```suggestion
    [IsMultiplicative P]
```

Reply:
great! This shortened `InducedWideCategory D F P hP` to `InducedWideCategory D F P` all throughout the file.

Reply:
and shortened `WideSubcategory P hP` to `WideSubcategory P`.